File: src/com/l2kt/gameserver/network/clientpackets/RequestTutorialQuestionMark.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.scripting.QuestState;

public class RequestTutorialQuestionMark extends L2GameClientPacket
{
	int _number;
	
	@Override
	protected void readImpl()
	{
		_number = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player player = getClient().getActiveChar();
		if (player == null)
			return;
		
		QuestState qs = player.getQuestState("Tutorial");
		if (qs != null)
			qs.getQuest().notifyEvent("QM" + _number + "", null, player);
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestTutorialQuestionMark.kt
package com.l2kt.gameserver.network.clientpackets

class RequestTutorialQuestionMark : L2GameClientPacket() {
    internal var _number: Int = 0

    override fun readImpl() {
        _number = readD()
    }

    override fun runImpl() {
        val player = client.activeChar ?: return

        val qs = player.getQuestState("Tutorial")
        qs?.quest?.notifyEvent("QM$_number", null, player)
    }
}